hs_core/hydroshare/resource.py

```diff
@@ -376,7 +382,24 @@ def create_resource(
         if not metadata:
             metadata = []
 
-        add_resource_files(resource.short_id, *files)
+        if len(files) == 1 and unpack_file and zipfile.is_zipfile(files[0]):
+            # Add contents of zipfile asynchronously; wait 30 seconds to be "sure" that resource creation
+            # has finished.
+            import logging
```

Comment:
Yes, agree the import should be at the top.  Also, by removing the logging I mean the informational debug messages.  Errors should be logged, but I don't want to litter the logs with things like "this file added to such and such a resource".